### PR TITLE
Update eslint infrastructure

### PR DESCRIPTION
**Pull request by loay**
- Update eslint version
- Update eslint-config-loopback version
- Fix linting errors.

### CODE REVIEW

**Comment by loay** [comment]
@rmg Hi Ryan,
did the server credentials change or do they need to be updated as they are all failing on NVM?
Thanks.

**Comment by rmg** [comment]
@loay the credentials aren't set in the cis-jenkins environment. I had been holding off to see which repos actually needed them. I'll get it sorted out.

In the mean time, you can ignore these failures since they are passing on the same platforms on the other CI (`loopback-connector-mssql/label=ubuntu-4` == `[cis-jenkins] nvm,4`).

File: lib/migration.js [hunk 1]
```diff
@@ -333,7 +334,7 @@ function mixinMigration(MsSQL) {
         } else if (idProp.type === String) {
           if (idProp.generated !== false) {
             sql.push(self.columnEscaped(model, modelPKID) +
-              ' [uniqueidentifier] DEFAULT newid() NOT NULL');
+              ' [varchar](64) DEFAULT newid() NOT NULL');
```
Comment (superkhau) [inline lib/migration.js]: is this a linting error? Seems like you're changing code here

**Comment by superkhau** [comment]
@loay LGTM except that one line. Seems like you're changing unrelated code there. I'll let you figure out that part. :ship:

**Comment by loay** [comment]
Removed the unrelated code but got to push this PR https://github.com/strongloop/loopback-connector-mssql/pull/92 first as ci will fail on this one now.

**Comment by loay** [comment]
Removed unrelated code

**Comment by jannyHou** [comment]
LGTM and I am looking into the failure.

**Comment by jannyHou** [comment]
LGTM 👍

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@slnode test please

**Pull request merged**